azure-storage-cpp-lite/include/storage_credential.h

```diff
@@ -43,27 +46,51 @@ namespace microsoft_azure {
             std::vector<unsigned char> m_account_key;
         };
 
-        class shared_access_signature_credential : public storage_credential {
+        class shared_access_signature_credential final : public storage_credential {
         public:
             shared_access_signature_credential(const std::string &sas_token)
                 : m_sas_token(sas_token) {
-		// If there is a question mark at the beginning of the sas token, erase it for easier processing in sign_request.
-		if (!m_sas_token.empty() && m_sas_token[0] == '?') {
-		    m_sas_token.erase(0, 1);
-		}
+            // If there is a question mark at the beginning of the sas token, erase it for easier processing in sign_request.
+            if (!m_sas_token.empty() && m_sas_token[0] == '?') {
+                m_sas_token.erase(0, 1);
+            }
 	    }
 
-            AZURE_STORAGE_API void sign_request(const storage_request_base &r, http_base &h, const storage_url &url, const storage_headers &headers) const override;
-            AZURE_STORAGE_API std::string transform_url(std::string url) const override;
+        AZURE_STORAGE_API void sign_request(const storage_request_base &r, http_base &h, const storage_url &url, const storage_headers &headers) const override;
+        AZURE_STORAGE_API std::string transform_url(std::string url) const override;
 
         private:
             std::string m_sas_token;
         };
 
-        class anonymous_credential : public storage_credential {
+        class anonymous_credential final : public storage_credential {
         public:
             void sign_request(const storage_request_base &, http_base &, const storage_url &, const storage_headers &) const override {}
         };
 
+        class token_credential : public storage_credential {
+        public:
+            AZURE_STORAGE_API token_credential();
+            // DEPRECATION NOTICE: This is mostly meant for compatibility with sample.cpp
+            // The new token credential uses the OAuthTokenCredentialManager, and explicitly set tokens disable this functionality!
+            AZURE_STORAGE_API token_credential(const std::string &token);
```

Comment:
TODO: Perhaps mark changes made within CPPlite for the sake of blobfuse in comments so that when it comes time to merge in a new version of CPPlite we can pretty easily decipher what needs to stay and what gets replaced